.changeset/tough-countries-behave.md

```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/auth-construct-alpha': patch
+---
+
+Auth config is now generated from resources instead of input props to allow customers to see their custom cdk overrides in the config.
```

Comment:
@edwardfoyle 

I spoke with @josefaidt the other day and we concluded that:

```
const { cfnUserPool } = backend.auth.resources.cfnResources

cfnUserPool.addPropertyOverride(
  "Policies.PasswordPolicy.MinimumLength",
  32
)
```

should just work and for CDK-level overridable props this is strategy we'd guide customers to (so that fronted config has higher chance to match deployed resources).

This:
```
backend.addOutput({ password_min_leght: 3});
```
should also work but it's a strategy recommended for "bring your own cognito/auth", not when they use `defineAuth`.

Attempt to use both together results in validation error as of today. (and doesn't seem to be valid use case today).

Reply:
Whether we have to use `Lazy` or not I'm not sure but we should have test for that.
Afaik data construct doesn't use `Lazy` and it seemed to be working when I tested that.